src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -11,12 +12,26 @@ internal class EndToEndTraceHelper
         private static string appName;
         private static string slotName;
 
-        // TODO: Replace TraceWriter with structured event client
-        private readonly TraceWriter appTraceWriter;
+        private readonly ILogger logger;
 
-        public EndToEndTraceHelper(TraceWriter appTraceWriter)
+        public EndToEndTraceHelper(JobHostConfiguration config)
         {
-            this.appTraceWriter = appTraceWriter;
+            string instrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+            string logCategory = this.GetType().Namespace;
+            if (!string.IsNullOrEmpty(instrumentationKey))
+            {
+                var filter = new LogCategoryFilter
+                {
+                    DefaultLevel = LogLevel.Debug
+                };
+
+                filter.CategoryLevels[logCategory] = LogLevel.Debug;
+
+                config.LoggerFactory = new LoggerFactory()
+                    .AddApplicationInsights(instrumentationKey, filter.Filter);
+            }
+
+            this.logger = config.LoggerFactory?.CreateLogger(logCategory);
```

Comment:
You shouldn't be overwriting the LoggerFactory -- instead, just use the one that's passed in. This is the only line you should need.